include/drivers/peci.h

```diff
@@ -0,0 +1,299 @@
+#define PECI_GET_TEMP_ERR_LSB_TEMP_LO  (0x2U)
+#define PECI_GET_TEMP_ERR_LSB_TEMP_HI  (0x3U)
+
+/** RdPkgConfig command format. */
+#define PECI_RD_PKG_WR_LEN             (5U)
+#define PECI_RD_PKG_LEN_BYTE           (2U)
+#define PECI_RD_PKG_LEN_WORD           (3U)
+#define PECI_RD_PKG_LEN_DWORD          (5U)
+#define PECI_RD_PKG_CMD_LEN            (8U)
+
+/** WrPkgConfig command format */
+#define PECI_WR_PKG_RD_LEN              (1U)
+#define PECI_WR_PKG_LEN_BYTE            (7U)
+#define PECI_WR_PKG_LEN_WORD            (8U)
+#define PECI_WR_PKG_LEN_DWORD           (10U)
+#define PECI_WR_PKG_CMD_LEN             (9U)
+
+/** RdIAMSR command format */
+#define PECI_RD_IAMSR_WR_LEN            (5U)
+#define PECI_RD_IAMSR_LEN_BYTE          (2U)
+#define PECI_RD_IAMSR_LEN_WORD          (3U)
+#define PECI_RD_IAMSR_LEN_DWORD         (5U)
+#define PECI_RD_IAMSR_LEN_QWORD         (9U)
+#define PECI_RD_IAMSR_CMD_LEN           (8U)
+
+/** WrIAMSR command format */
+#define PECI_WR_IAMSR_RD_LEN            (1U)
+#define PECI_WR_IAMSR_LEN_BYTE          (7U)
+#define PECI_WR_IAMSR_LEN_WORD          (8U)
+#define PECI_WR_IAMSR_LEN_DWORD         (10U)
+#define PECI_WR_IAMSR_LEN_QWORD         (14U)
+#define PECI_WR_IAMSR_CMD_LEN           (9U)
+
+/** RdPCIConfig command format */
+#define PECI_RD_PCICFG_WR_LEN           (6U)
+#define PECI_RD_PCICFG_LEN_BYTE         (2U)
+#define PECI_RD_PCICFG_LEN_WORD         (3U)
+#define PECI_RD_PCICFG_LEN_DWORD        (5U)
+#define PECI_RD_PCICFG_CMD_LEN          (9U)
+
+/** WrPCIConfig command format */
+#define PECI_WR_PCICFG_RD_LEN           (1U)
+#define PECI_WR_PCICFG_LEN_BYTE         (8U)
+#define PECI_WR_PCICFG_LEN_WORD         (9U)
+#define PECI_WR_PCICFG_LEN_DWORD        (11U)
+#define PECI_WR_PCICFG_CMD_LEN          (10U)
+
+/** RdPCIConfigLocal command format */
+#define PECI_RD_PCICFGL_WR_LEN          (5U)
+#define PECI_RD_PCICFGL_RD_LEN_BYTE     (2U)
+#define PECI_RD_PCICFGL_RD_LEN_WORD     (3U)
+#define PECI_RD_PCICFGL_RD_LEN_DWORD    (5U)
+#define PECI_RD_PCICFGL_CMD_LEN         (8U)
+
+/** WrPCIConfigLocal command format */
+#define PECI_WR_PCICFGL_RD_LEN          (1U)
+#define PECI_WR_PCICFGL_WR_LEN_BYTE     (7U)
+#define PECI_WR_PCICFGL_WR_LEN_WORD     (8U)
+#define PECI_WR_PCICFGL_WR_LEN_DWORD    (10U)
+#define PECI_WR_PCICFGL_CMD_LEN         (9U)
+
+/**
+ * @brief PECI buffer structure
+ *
+ * @param buf is a valid pointer on a data buffer, or NULL otherwise.
+ * @param len is the length of the data buffer expected to received without
+ * considering the frame check sequence byte.
+ *
+ * Note: Frame check sequence byte is added into rx buffer, need to allocate
+ * an additional byte for this in rx buffer.
+ */
+struct peci_buf {
+	u8_t *buf;
+	size_t len;
+};
+
+/**
+ * @brief PECI transaction packet format.
+ */
+struct peci_msg {
+	/** Client address */
+	u8_t addr;
+	/** Command code */
+	enum peci_command_code cmd_code;
+	/** Pointer to buffer of write data */
+	const struct peci_buf *tx_buffer;
```

Comment:
why does tx_buffer and rx_buffer have to be a pointer?
can't you make this much simpler and directly embed them?